edit/bzlmod/bzlmod.go

```diff
@@ -432,3 +591,89 @@ func collectApparentNamesAndIncludes(f *build.File) (map[string]string, []string
 
 	return apparentNames, includeLabels
 }
+
+// parseRepoMapping parses a repo string which may be in the form "key=value" or just "value".
+// Returns (key, value, isMapping) where isMapping indicates if it was a mapping.
+func parseRepoMapping(repo string) (key, value string, isMapping bool) {
+	key, value, found := strings.Cut(repo, "=")
+	if found {
+		return key, value, true
+	}
+	return "", repo, false
+}
+
+// starlarkReservedKeywords contains Starlark reserved keywords that cannot be used as identifiers.
+var starlarkReservedKeywords = map[string]bool{
+	"and":      true,
+	"break":    true,
+	"continue": true,
+	"def":      true,
+	"elif":     true,
+	"else":     true,
+	"for":      true,
+	"if":       true,
+	"in":       true,
+	"lambda":   true,
+	"load":     true,
+	"not":      true,
+	"or":       true,
+	"pass":     true,
+	"return":   true,
+	"while":    true,
+	"False":    true,
+	"None":     true,
+	"True":     true,
+}
+
+// isValidIdentifier checks if a string is a valid Starlark identifier.
+// Per Starlark rules: must start with letter or underscore, followed by letters, digits, or underscores,
+// and must not be a reserved keyword.
+func isValidIdentifier(s string) bool {
+	if len(s) == 0 {
+		return false
+	}
+
+	// Check if it's a reserved keyword
+	if starlarkReservedKeywords[s] {
+		return false
+	}
+
+	for i, r := range s {
+		if i == 0 {
+			// First character must be letter or underscore
+			if !unicode.IsLetter(r) && r != '_' {
```

Comment:
At least Bazel's Starlark interpreter only allows ASCII characters in identifiers, not arbitrary Unicode letters: https://cs.opensource.google/bazel/bazel/+/d61a021aa63739f28ae1ba9f00ecf0c05e3edd09:src/main/java/net/starlark/java/syntax/Identifier.java;l=86